client/scripts/views/components/admin_panel.ts

```diff
@@ -0,0 +1,634 @@
+import m, { Vnode } from 'mithril';
+import $ from 'jquery';
+import { OffchainThread, OffchainTag, CommunityInfo, RolePermission, ChainInfo, ChainNetwork, RoleInfo } from 'models';
+import { Button, Classes, Dialog, Icon, Icons, Tag, TagInput, ListItem, Table, Input, List, TextArea, Switch, Tabs, TabItem, RadioGroup, Form, FormGroup, FormLabel } from 'construct-ui';
+import app from 'state';
+import { sortAdminsAndModsFirst } from 'views/pages/discussions/roles';
+import User from './widgets/user';
+import 'components/admin_panel.scss';
+
+const RoleRow: m.Component<{ roledata?, onRoleUpdate?: Function }> = {
```

Comment:
Can we break this file out into an `admin` component directory and give each sizable component its own file?